base/Base.jl

```diff
@@ -196,6 +197,7 @@ end
 
 # core operations & types
 include("promotion.jl")
+include("tuple_from_iterator.jl")
```

Comment:
Does this really need to exist this early? The code would be a lot cleaner if it could be moved to after `int` (so you can use `+`)